src/state/slices/opportunitiesSlice/resolvers/osmosis/utils.ts

```diff
@@ -0,0 +1,200 @@
+      assetNamespace: ASSET_NAMESPACE.native,
+      assetReference: denom,
+      chainId: osmosisChainId,
+    })
+  }
+
+  if (denom.startsWith('ibc')) {
+    return toAssetId({
+      assetNamespace: ASSET_NAMESPACE.ibc,
+      assetReference: denom.split('/')[1],
+      chainId: osmosisChainId,
+    })
+  }
+
+  return toAssetId({
+    assetNamespace: ASSET_NAMESPACE.slip44,
+    assetReference: ASSET_REFERENCE.Osmosis,
+    chainId: osmosisChainId,
+  })
+}
+
+export const getPools = async (): Promise<OsmosisPool[]> => {
+  try {
+    /* Fetch Osmosis pool data */
+    const { data: poolData } = await memoize(async (): Promise<AxiosResponse<OsmosisPoolList>> => {
+      return await axios.get<OsmosisPoolList>(
+        (() => {
+          const url = new URL('gamm/v1beta1/pools', getConfig().REACT_APP_OSMOSIS_LCD_BASE_URL)
+          url.searchParams.set(
+            'pagination.limit',
+            getConfig().REACT_APP_OSMOSIS_POOL_PAGINATION_LIMIT.toString(),
+          )
+          return url.toString()
+        })(),
+      )
+    })()
+
+    if (!poolData) throw new Error('Unable to fetch Osmosis liquidity pool metadata')
+
+    /* Fetch historical data for Osmosis pools */
+    const { data: historicalDataByPoolId } = await memoize(async (): Promise<any> => {
+      return await axios.get<PoolHistoricalDataList>(
+        (() => {
+          const url = new URL('pools/v2/all', getConfig().REACT_APP_OSMOSIS_IMPERATOR_BASE_URL)
+          url.searchParams.set(
+            'low_liquidity',
+            getConfig().REACT_APP_OSMOSIS_ALLOW_LOW_LIQUIDITY_POOLS.toString(),
+          )
+          return url.toString()
+        })(),
+      )
+    })()
+
+    if (!historicalDataByPoolId)
+      throw new Error('Unable to fetch historical data for Osmosis liquidity pools')
+
+    const getPoolTVL = (pool: OsmosisBasePool): string => {
+      const marketData = historicalDataByPoolId[pool.id]
+      return bnOrZero(marketData[0].liquidity).toString()
+    }
+
+    const calculatePoolAPY = (pool: OsmosisBasePool): string => {
+      const poolHistoricalData = historicalDataByPoolId[pool.id][0] // Identical pool-level historical data exists on both asset entries
+
+      /** Pool fee data is represented in API response like '0.2%', so we strip off the percent sign
+       * then multiply the remaining number by 0.01 to get the fee multiplier.
+       * Ex: (2% of X) = X * 2 * 0.01
+       */
+      const feeMultiplier = bnOrZero(poolHistoricalData.fees.split('%')[0]).multipliedBy(bn(0.01))
+      const feesSpent7d = bnOrZero(poolHistoricalData.volume_7d).multipliedBy(feeMultiplier)
+      const averageDailyFeeRevenue = feesSpent7d.dividedBy(bn(7))
+      const annualRevenue = averageDailyFeeRevenue.multipliedBy(bn(365))
+      const poolTVL = bnOrZero(getPoolTVL(pool))
+
+      if (poolTVL.eq(0) || annualRevenue.eq(0)) return bn(0).toString()
+
+      return annualRevenue.dividedBy(poolTVL).multipliedBy(bn(100)).toString()
```

Comment:
APY should be in 0. notation, see current opportunities for reference:

<img width="204" alt="image" src="https://user-images.githubusercontent.com/17035424/212911952-41bf0a2b-8bba-4921-bd4a-547f7b5998c7.png">

```suggestion
      return annualRevenue.dividedBy(poolTVL).toString()
```